data-processing/tests/test_extract_definitions.py

```diff
@@ -0,0 +1,110 @@
+    assert len(symbol_nickname_pairs) == len(gold), "Incorrect number of Symbol-Nicknames"
+    for prediction_pair, gold_pair in zip(symbol_nickname_pairs, gold):
+        if symbol_texs is not None and prediction_pair.term_start in symbol_texs:
+            assert symbol_texs[prediction_pair.term_start] == gold_pair[0], "Symbol Incorrect"
+        else:
+            assert text[prediction_pair.term_start:prediction_pair.term_end] == gold_pair[0], "Symbol Incorrect"
+        assert text[prediction_pair.definition_start:prediction_pair.definition_end] == gold_pair[1], "Nickname Incorrect"
+        
+def test_abbreviation_expansion_pairs(text, gold):
+    featurized_text = model.featurize(text)
+    features = [featurized_text]
+    abbreviation_pairs = get_abbreviation_pairs(
+        text, features[0], model.nlp
+    )
+    assert len(abbreviation_pairs) == len(gold), "Incorrect number of Abbreviations"
+    for prediction_pair, gold_pair in zip(abbreviation_pairs, gold):
+        assert text[prediction_pair.term_start:prediction_pair.term_end] == gold_pair[0], "Abbreviation Incorrect"
+        assert text[prediction_pair.definition_start:prediction_pair.definition_end] == gold_pair[1], "Expansion Incorrect"
+    
+
+def test_basic():
+    # Test Abbreviation-Expansion detection
+    ## Case 1: Abbreviation in parenthesis
+    text = "We use a Convolutional Neural Network (CNN) based architecture in this model, which is an improvement over state-of-the-art"
+    gold = [('CNN','Convolutional Neural Network')]
+    test_abbreviation_expansion_pairs(text, gold)
+
+    # These two are important cases that don't yet work in the current abbreviation detection system
+    # ## Case 2: Abbreviation not in parenthesis
+    # text = "We use a Convolutional Neural Network, known as CNN, based architecture in this model, which is an improvement over state-of-the-art"
+    # gold = [('CNN','Convolutional Neural Network')]
+    # test_abbreviation_expansion_pairs(text, gold)
+
+    # ## Case 3: Expansion in parenthesis
+    # text = "GANs (Generative Adversarial networks) outperform most generative models in the novel human face generation task."
+    # gold = [(GANs,Generative Adversarial networks)]
+    # test_abbreviation_expansion_pairs(text, gold)
+
+    ## Case 4: Abbreviation contains multiple lowercase letters
+    text = "We propose a new class of architectures called Conductive Networks (CondNets) in this paper."
+    gold = [('CondNets','Conductive Networks')]
+    test_abbreviation_expansion_pairs(text, gold)
+
+    # Test Symbol-Nickname detection
+    ## Case 1: Nickname before symbol 
+    text = "The agent acts with a policy SYMBOL in each timestep SYMBOL."
+    tex = "The agent acts with a policy [[FORMULA:\pi]] in each timestep [[FORMULA:t]]"
+    gold = [('\pi','policy'),('t','timestep')]
+    test_symbol_nickname_pairs(text, tex, gold)
+    
+    ## Case 2: Nickname after symbol 
```

Comment:
Let's split each of these cases into separate tests so that the test runner can more precisely pinpoint what cases are failing in its summaries.

Reply:
I went through the pytest documentation and handled it such that each each example is a new function call